### PR TITLE
Make release, reset methods of Lock and reset_all function delete signal keys via additional call to Redis

### LINKED ISSUE #47: Replace expiration with explicit deletion
Refer to this [comment][1]

[1]: https://github.com/ionelmc/python-redis-lock/pull/32#issuecomment-183056293

### PR DESCRIPTION
**Pull request by AndrewPashkin**
Subj.

Closes  #47.

### CODE REVIEW

File: src/redis_lock/__init__.py [hunk 1]
```diff
@@ -53,14 +51,24 @@
...
 """
 
 RESET_ALL_SCRIPT_HASH = sha1(RESET_ALL_SCRIPT).hexdigest()
 
+DELETE_ALL_SIGNAL_KEYS_SCRIPT = b"""
+    local signals = redis.call('keys', 'lock-signal:*')
+    for _, signal in pairs(signals) do
+        redis.call('del', signal)
+    end
+    return #signals
```
Comment (ionelmc) [inline src/redis_lock/__init__.py]: Does this actually need to return anything?

**Comment by ionelmc** [comment]
Ooops, seems I forgot about this. I hope it's still fresh in your head :-)

**Comment by AndrewPashkin** [comment]
I'll retrigger build.

**Comment by ionelmc** [comment]
No need. Restarted it now. Not sure if it merges correctly with master. Maybe do a rebase?

**Comment by AndrewPashkin** [comment]
>  Maybe do a rebase?
Done

**Comment by AndrewPashkin** [comment]
:shipit:

**Comment by ionelmc** [comment]
Alright. Thanks!

**Comment by ionelmc** [comment]
@AndreiPashkin fyi this unfortunately will get reverted to fix #71.

**Pull request merged**